src/lib/reasoners/adt_rel.ml

```diff
@@ -373,20 +440,15 @@ let add_aux env (uf:uf) (r:r) t =
       { env with seen_access = SE.add t env.seen_access }
 
     | Sy.Op Sy.Destruct _, _ ->
-      assert false (* not possible *)
+      (* This case is excluded by the parser. *)
```

Comment:
I am not sure I understand what the parser has to do with this, can you clarify?

Reply:
Technically, it's not an invariant of the parser but the arity of the operator `Destruct` is `1`. I suggest: 
```ocaml
(* The arity of the [Sy.Destruct] operator is 1. *)
```

Currently, we can produce a destructor with any number of arguments because the internal structure of expressions isn't sufficiently opaque. You can produce such terms with `Expr.mk_term` for instance. Builtin operators shouldn't be build by `mk_term` in fact. We should reserve `mk_term` for uninterpreted functions.